mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -59,28 +70,66 @@ class ExperimentView extends Component {
     runsSelected: {},
     paramKeyFilterInput: '',
     metricKeyFilterInput: '',
+    lifecycleFilterInput: LIFECYCLE_FILTER.ACTIVE,
     searchInput: '',
     searchErrorMessage: undefined,
     sort: {
       ascending: false,
       isMetric: false,
       isParam: false,
       key: "start_time"
-    }
+    },
+    showDeleteRunModal: false,
+    showRestoreRunModal: false,
   };
 
+  shouldComponentUpdate(nextProps, nextState) {
+    // Don't update the component if a modal is showing before and after the update try.
+    if (this.state.showDeleteRunModal && nextState.showDeleteRunModal) return false;
+    if (this.state.showRestoreRunModal && nextState.showRestoreRunModal) return false;
+    return true;
+  }
+
+
   static getDerivedStateFromProps(nextProps, prevState) {
-    const { searchInput, paramKeyFilter, metricKeyFilter } = nextProps;
+    // Compute the actual runs selected. (A run cannot be selected if it is not passed in as a
+    // prop)
+    const newRunsSelected = {};
```

Comment:
To confirm: is the intent of this to preserve which runs are selected e.g. after a search? It for some reason doesn't seem to work for me

Reply:
No, it's actually to remove "deleted" runs from the selected set after a deletion happens.